src/material/timepicker/timepicker.html

```diff
@@ -11,6 +11,7 @@
     @for (option of _timeOptions; track option.value) {
       <mat-option
         [value]="option.value"
+        [disabled]="option.disabled"
```

Comment:
My thinking with the timepicker was actually to just not include the disabled items in the list. With something like this we'd also have to handle a bunch of cases like "what happens if the user manually types the value of a disabled option?"